File: Binary_Trees/111-minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if root == None:
            return 0

        l = self.minDepth(root.left)
        r = self.minDepth(root.right)

        if root.left == None:
            return 1 + r
        if root.right == None:
            return 1 + l

        return 1 + min(l, r)

File: Binary_Trees/111-minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if(root == NULL) return 0;

        int l = minDepth(root->left);
        int r = minDepth(root->right);

        if(root->left == NULL) return 1 + r;
        if(root->right == NULL) return 1 + l;

        return 1 + min(l, r);
    }
};